mbzirc_ign/src/GameLogicPlugin.cc

```diff
@@ -863,66 +829,7 @@ void GameLogicPlugin::PostUpdate(
   // triggers the start signal.
   if (!this->dataPtr->started)
   {
-    _ecm.Each<gazebo::components::Sensor,
-              gazebo::components::ParentEntity>(
-        [&](const gazebo::Entity &_entity,
-            const gazebo::components::Sensor *,
-            const gazebo::components::ParentEntity *_parent) -> bool
-        {
-          // Get the model. We are assuming that a sensor is attached to
-          // a link.
-          auto parent = _ecm.Component<gazebo::components::ParentEntity>(
-              _parent->Data());
-          auto model = parent;
-
-          // find top level model
-          while (parent && _ecm.Component<gazebo::components::Model>(
-                 parent->Data()))
-          {
-            model = parent;
-            parent = _ecm.Component<gazebo::components::ParentEntity>(
-                parent->Data());
-          }
-
-          if (model)
-          {
-            // \todo(anyone)
-            // Check if the robot has beyond the staging area.
-            // we need to trigger the /mbzirc/start.
-
-            // Get the model name
-            Entity entity = model->Data();
-            auto mName =
-              _ecm.Component<gazebo::components::Name>(entity);
-            if (this->dataPtr->robotNames.find(entity) ==
-                this->dataPtr->robotNames.end())
-            {
-              this->dataPtr->robotNames[entity] = mName->Data();
-              this->dataPtr->robotInitialPos[entity] =
-                  _ecm.Component<components::Pose>(entity)->Data().Pos();
-
-              // Subscribe to battery state in order to log battery events.
-              std::string batteryTopic = std::string("/model/") +
-                mName->Data() + "/battery/linear_battery/state";
-              this->dataPtr->node.Subscribe(batteryTopic,
-                  &GameLogicPluginPrivate::OnBatteryMsg, this->dataPtr.get());
-            }
-
-            // store camera / rgbd camera sensor
-            // later user for target confirmation in image stream
-            auto camComp = _ecm.Component<gazebo::components::Camera>(_entity);
-            if (camComp)
-            {
-              this->dataPtr->cameraSensors.insert(_entity);
-            }
-            auto rgbdComp = _ecm.Component<gazebo::components::RgbdCamera>(_entity);
-            if (rgbdComp)
-            {
-              this->dataPtr->cameraSensors.insert(_entity);
-            }
-          }
-          return true;
```

Comment:
we are storing camera sensors that'll be used later on for video stream validation. I think this logic is also gone after refactor.

Reply:
[284d26d](https://github.com/osrf/mbzirc/pull/164/commits/284d26d66199477109fbd1d3d651a74d477246a2)